### DISCUSSION TITLE
Missing date/dateTime etc. in data

### DISCUSSION THREAD

**Discussion by miekeGR** [ternaustralia/ontology_tern#161] [Modelling]
Greetings TERN folks,

Through creating data models reflecting data the BDR is expecting to get from WA BIO, we have run into some situations where the data we require is missing. I have requested a few things to accommodate lacking data such as some generic procedures and relaxing the restrictions on sosa:resultTime. But I have a new concern regarding the requirement to have a date/time on every Sampling/Observation. When I say that it sounds silly, because of course every Sampling and Observation occur at a point in time so it makes sense that a resultDateTime is mandatory, but I want to discuss these:

tern:resultDateTime
tern:Sampling and tern:Observation both require a tern:resultDateTime (or sosa:resultTime once the datatype constraints are loosened) the result was recorded, whether in situ or at a later point of the phenomenon being observed. We can generally find the equivalent of dwc:eventDate in the datasets, which we can use for field sampling dates, but knowing how to handle tern:resultDateTime for sub-samplings and tern:Observations is not always straightforward. For example, sometimes we find the equivalent of dwc:dateIdentified to help populate tern:resultDateTime for a taxon determination observation; however, generally, this information is not explicit. Recording other observations such as life stage or sex may also not be determined in the field, but rather at some time later in the lab so we cannot assume the date of the observation was the same as the dwc:eventDate.

sosa:phenomenonTime
tern:Observations require a time that the phenomenon occurred (as opposed to the date/time the result of the Observation was recorded). To be accurate in providing this information is not always doable (e.g. a pitfall trap left out for several days/weeks cannot give accuracy as to which time the contents were captured) and often not recorded or transferred in datasets we are seeing. For now, I have been assuming dwc:eventDate will have to be good enough.

I would like to know what you think about dateTime assumptions.

**Comment by edmondchuc** [comment]
Hi @miekeGR, thanks for your examples. 

I think this is a very difficult problem and there may not be a universally accepted solution. My first instinct is to either change the data or change what is required in the data model. We can't change the data or the survey data collection protocol since it has already been collected but I'm also not sure if changing the data model is the correct solution either. 

In most cases with the datasets we've looked at (AusPlots Rangelands and CORVEG), we have followed the same assumptions as you. Date time information for individual samplings or observations are not explicitly recorded in the dataset. We generally use the same value from the site visit. This is not perfect as it limits the kind of insights we can get from the data but it's the best we can do without violating the requirements of the data model.

Now that we have `tern:resultDateTime`, we can drop the time portion for samplings and observations and only use the date portion. In most cases this would be accurate for ecological surveys. The samplings and observations occur sometime during the date in which the site visit occurred. Though this may not be true for all samplings and observations as you have mentioned in regards to life stage or sex. There will also be issues with this approach when site visits span over multiple days and the data does not differentiate which day samplings or observations were made.

I think the best we can do is record whatever assumptions we have made about the data and ensure it is available in the metadata to ensure transparency. 

@dr-shorthair do you have any experiences or tips you would like to share?

@habacucfm from an ecological survey point of view, do you have any comments on the importance of date time for samplings and observations?

**Reply by miekeGR** [reply]
Thanks @edmondchuc that's interesting to hear you have faced similar situations. With the AusPlots Rangelands and CORVEG, how do you approach documenting the assumptions?

**Comment by dr-shorthair** [comment]
You can't make stuff up. But certainly you can make intelligent inferences from information elsewhere in the dataset or context. 

While `sosa:resultTime` is an instant, `sosa:phenomenonTime` allows for a `time:TemporalEntity` which permits an interval, which can be single-ended. So I think you can do it right. 

However, this may result in a whole load of `time:ProperInterval` in the graph store, so queries focussing on time will have to be able to deal with those.

**Comment by habacucfm** [comment]
I would tend to think that time (very specific time) matters little, unless you are making physiological observations (think respiration) or animal behaviour. Most other processes that we tend to observe happen over periods of time such that a simple date or even year and month or knowing that the survey time started on year X and continued  on almost yearly basis for N years will suffice.
